test/test-parse-expression.cpp

```diff
@@ -3161,6 +3161,273 @@ TEST_F(test_parse_expression,
   }
 }
 
+TEST(test_parse, test_expression_for_potential_side_effects) {
+  {
+    test_parser p(u8"class {};"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"typeof 42;"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_FALSE(side_effects);
+  }
+
+  {
+    test_parser p(u8"typeof f();"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"await foo;"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"import foo;"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"++x;"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"x++;"_sv);
+    expression* ast = p.parse_expression();
+    bool side_effects = p.parser().has_potential_side_effects(ast);
+    EXPECT_TRUE(side_effects);
+  }
+
+  {
+    test_parser p(u8"...foo"_sv);
```

Comment:
Blocking: This is a syntax error. Spread isn't allowed on its own. I think you meant one of the following:
```js
{...foo}
```
```js
[...foo]
```
```js
f(...foo)
```